src/main/java/org/snomed/snowstorm/core/data/services/RelationshipService.java

```diff
@@ -46,9 +50,56 @@
 	@Autowired
 	private ConceptUpdateHelper conceptUpdateHelper;
 
+	@Autowired
+	private MRCMLoader mrcmLoader;
+
 	public Relationship findRelationship(String branchPath, String relationshipId) {
-		Page<Relationship> relationships = findRelationships(branchPath, relationshipId, null, null, null, null, null, null, null, null, PageRequest.of(0, 1));
-		return relationships.getTotalElements() > 0 ? relationships.getContent().get(0) : null;
+		final Page<Relationship> relationships = findRelationships(branchPath, relationshipId, null, null, null, null, null, null, null, null, PageRequest.of(0, 1));
+		if (relationships.isEmpty()) {
+			return null;
+		}
+
+		final Relationship relationship = relationships.getContent().get(0);
+		setConcreteValueFromMRCM(branchPath, relationship);
+
+		return relationship;
+	}
+
+	/**
+	 * Set the ConcreteValue of each Relationship.
+	 *
+	 * @param branchPath    The branchPath to load active MRCM data from.
+	 * @param relationships The Relationships to update.
+	 * @throws RuntimeServiceException When there is an issue reading MRCM.
+	 */
+	public void setConcreteValueFromMRCM(String branchPath, Relationship... relationships) {
+		final BranchCriteria branchCriteria = versionControlHelper.getBranchCriteria(branchPath);
```

Comment:
This method is cheap-ish but not free. We will be creating a new branch criteria for every inferred relationship when loading a concept even if it's not used because the MRCM is cached. It would be better to just pass the branch path to the MRCM loader because usually the MRCM will be cached and the branch criteria will not be needed. The mrcm loader will need access to the versionControlHelper so it can create a BranchCriteria if needed.